modules/sdk/src/main/scala/org/tessellation/sdk/app/TessellationIOApp.scala

```diff
@@ -205,6 +220,23 @@ abstract class TessellationIOApp[A <: CliMethod](
       }
     }
 
+  private def getMajorityPeerIds(
+    prioritySeedlist: Option[Set[SeedlistEntry]],
+    defaultMajorityPeerIds: Option[NonEmptySet[PeerId]],
+    env: AppEnvironment
+  ): IO[Option[NonEmptySet[PeerId]]] =
+    prioritySeedlist
+      .map(_.map(_.peerId))
+      .flatMap(_.toSortedSet.toNes)
+      .orElse(defaultMajorityPeerIds)
+      .pure[IO]
+      .flatTap(ids => logger.info(s"Majority Peer IDs $ids"))
+      .flatTap {
+        MonadThrow[IO]
+          .fromOption(_, new IllegalStateException("Majority Peer IDs required"))
+          .whenA(env =!= AppEnvironment.Dev)
+      }
```

Comment:
can you make sure that throw in `flatTap` also stops the flow? it's kinda counterintuitive as `flatTap` should not influence the value so whatever is executed here should be "within own bubble". but I may be wrong

Reply:
@marcinwadon This test will fail with an error in `IO` -- is that what you wanted confirmed?
```
test("throw in flatTap") { _ =>
  IO.pure(none[Int])
    .flatTap(MonadThrow[IO].fromOption(_, new Exception("boom")))
    .as(success)
}
```

Reply:
I don't think `flatTap` is intended to be a self-contained environment. The only difference between between that and `flatMap` is the return type.
"Apply a monadic function and discard the result while keeping the effect."

Reply:
@marcinwadon I found a cleaner way to do this which I think also conveys the intent better:
```
.reject { case None if env =!= AppEnvironment.Dev =>
  new IllegalStateException("Majority Peer IDs required")
}
```